mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -2297,3 +2298,40 @@ FailureOr<AffineValueMap> mlir::affine::simplifyConstrainedMinMaxOp(
   affine::canonicalizeMapAndOperands(&newMap, &newOperands);
   return AffineValueMap(newMap, newOperands);
 }
+
+Block *mlir::affine::findInnermostCommonBlockInScope(Operation *a,
+                                                     Operation *b) {
+  Region *aScope = mlir::affine::getAffineScope(a);
+  Region *bScope = mlir::affine::getAffineScope(b);
+  if (aScope != bScope)
+    return nullptr;
+
+  // Get the block ancestry of `a` while stopping at the affine scope.
+  auto getBlockAncestry = [&](Operation *op,
+                              SmallVectorImpl<Block *> &ancestry) {
+    Operation *curOp = op;
+    do {
```

Comment:
Should you do an assert that curOp is not null before the do-while loop as it will execute the body atleast once irrespective of whether the while condition holds? Or else you can use a regular while loop

Reply:
This is meant to be executed at least once - so a do-while loop. `curOp` is initialized with a function argument. The caller is expected to pass a non-null operation. We don't assert for such bugs.